lib/checkForNewCodeowner.js

```diff
@@ -0,0 +1,139 @@
+ */
+
+
+const utilsModule = require('./utils');
+const { oppiaMaintainers } = require('../userWhitelist.json');
+const CODE_OWNER_FILE_NAME = '.github/CODEOWNERS';
+
+/**
+ * This function returns all the new codeowners added to the codeowners file.
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ * @returns {string[]}
+ */
+const getNewCodeOwners = (file) => {
+  const newLine = '\n';
+  const newAddition = '+';
+  const changesArray = file.patch.split(newLine);
+  const newAdditions = changesArray.filter((change) => {
+    return change.startsWith(newAddition);
+  });
+
+  /**
+   * @type {string[]} addedCodeOwners
+   */
+  let addedCodeOwners = [];
+  newAdditions.forEach((addition) => {
+    const usernames = addition.split('@');
+    // The first item of the array will be the path to the file, and should
+    // be removed to get only the usernames.
+    usernames.shift();
+    addedCodeOwners.push(...usernames);
+  });
+  // Trim and add @ to the front of the username.
+  addedCodeOwners = addedCodeOwners.map((username) => '@' + username.trim());
+
+  // Remove all duplicates and return the array.
+  return Array.from(new Set(addedCodeOwners));
+};
+
+/**
+ * This function checks if a new codeowner has been added to the codeowner
+ * file and responds as required.
+ *
+ * @param {import('probot').Context} context
+ */
+const checkForNewCodeowner = async (context) => {
+  /**
+   * @type {import('probot').Octokit.PullsGetResponse} pullRequest
+   */
+  const pullRequest = context.payload.pull_request;
+
+  const changedFiles = await utilsModule.getAllChangedFiles(context);
+
+  const changedCodeOwnerFile = changedFiles.find(
+    (file) => file.filename === CODE_OWNER_FILE_NAME
+  );
+
+  if (changedCodeOwnerFile) {
+    const mainCodeOwnerFile = await utilsModule.getMainCodeOwnerfile();
+
+    const newlyAddedCodeOwners = getNewCodeOwners(changedCodeOwnerFile);
+
+    // Remove all exisiting codeowners.
+    const newCodeOwners = newlyAddedCodeOwners.filter(
+      (user) => !mainCodeOwnerFile.includes(user)
+    );
+    console.log({
+      newlyAddedCodeOwners,
+      newCodeOwners
+    })
+    if (newCodeOwners.length > 0) {
+      const changelogLabel = pullRequest.labels.find((label) =>
+        label.name.toUpperCase().startsWith('PR CHANGELOG')
+      );
+
+      let codeOwnerString = '';
+      if (newCodeOwners.length > 1) {
+        codeOwnerString =
+          'the following new code owners ' + newCodeOwners.join(', ');
+      } else {
+        codeOwnerString = 'a new code owner, ' + newCodeOwners[0];
+      }
+
+      let commentBody = '';
+      if (changelogLabel !== undefined) {
+        // Ping project owner.
+        const projectOwner = utilsModule.getProjectOwnerFromLabel(
+          changelogLabel.name.trim()
+        );
+
+        commentBody =
+          'Hi @' +
+          projectOwner +
+          ', this PR adds ' +
+          codeOwnerString +
+          '. Please take a look /cc @' +
```

Comment:
We can skip adding the core-maintainers. Just one of the reviewers should be fine. Also, add a comment to explain that we ping the project owner if changelog label is present otherwise we ping one of the reviewers just to flag the issue. 

Also, is it possible to link to the lines where new codeowners are added or add comment on all those lines as you were trying for tslint check?

Reply:
> We can skip adding the core-maintainers.

Done.


> Also, is it possible to link to the lines where new codeowners are added or add comment on all those lines as you were trying for tslint check?

No, we won't be able to get the exact line that got modified since it is not part of the payload.

Reply:
How were you doing that for the tslint check then?

Reply:
The tslint check was similar to that of the new jobs and new models where the bot comments and give a link to the file.

Reply:
Ah ok,  thanks for explaining.